src/coreComponents/constitutive/fluid/multifluid/compositional/models/ConstantViscosity.hpp

```diff
@@ -0,0 +1,129 @@
+                           ComponentProperties const & componentProperties ):
+    FunctionBaseUpdate( componentMolarWeight,
+                        componentProperties )
+  {}
+
+  template< int USD1 >
+  GEOS_HOST_DEVICE
+  void compute( real64 const & pressure,
+                real64 const & temperature,
+                arraySlice1d< real64 const, USD1 > const & phaseComposition,
+                real64 & viscosity,
+                bool useMass ) const;
+
+  template< int USD1, int USD2, int USD3 >
+  GEOS_HOST_DEVICE
+  void compute( real64 const & pressure,
+                real64 const & temperature,
+                arraySlice1d< real64 const, USD1 > const & phaseComposition,
+                arraySlice2d< real64 const, USD2 > const & dPhaseComposition,
+                real64 & viscosity,
+                arraySlice1d< real64, USD3 > const & dViscosity,
+                bool useMass ) const;
+};
+
+class ConstantViscosity : public FunctionBase
+{
+public:
+  ConstantViscosity( string const & name,
+                     array1d< string > const & componentNames,
+                     array1d< real64 > const & componentMolarWeight,
+                     ComponentProperties const & componentProperties );
+
+  virtual ~ConstantViscosity() override = default;
+
+  static string catalogName() { return ""; }
+
+  FunctionType functionType() const override
+  {
+    return FunctionType::VISCOSITY;
+  }
+
+  /// Type of kernel wrapper for in-kernel update
+  using KernelWrapper = ConstantViscosityUpdate;
+
+  /**
+   * @brief Create an update kernel wrapper.
+   * @return the wrapper
+   */
+  KernelWrapper createKernelWrapper() const;
+};
+
+template< int USD1 >
+GEOS_HOST_DEVICE
+GEOS_FORCE_INLINE
+void ConstantViscosityUpdate::compute( real64 const & pressure,
+                                       real64 const & temperature,
+                                       arraySlice1d< real64 const, USD1 > const & phaseComposition,
+                                       real64 & viscosity,
+                                       bool useMass ) const
+{
+  GEOS_UNUSED_VAR( pressure, temperature, useMass );
+  GEOS_UNUSED_VAR( phaseComposition );
+
+  viscosity = 0.001;
+}
+
+template< int USD1, int USD2, int USD3 >
+GEOS_HOST_DEVICE
+GEOS_FORCE_INLINE
+void ConstantViscosityUpdate::compute( real64 const & pressure,
+                                       real64 const & temperature,
+                                       arraySlice1d< real64 const, USD1 > const & phaseComposition,
+                                       arraySlice2d< real64 const, USD2 > const & dPhaseComposition,
+                                       real64 & viscosity,
+                                       arraySlice1d< real64, USD3 > const & dViscosity,
+                                       bool useMass ) const
+{
+  GEOS_UNUSED_VAR( pressure, temperature, useMass );
+  GEOS_UNUSED_VAR( phaseComposition, dPhaseComposition );
+
+  viscosity = 0.001;
```

Comment:
Hardcoded value?

Reply:
The actual implementation in the static functions is not "linked" yet. So all values are hardcoded constants - to ensure that the PVTDriver runs and outputs these values. The idea was to show the design layout and then, in subsequent PRs, link to the actual implementation.